lnst/Tests/Iperf.py

```diff
@@ -34,57 +35,52 @@ def run(self):
 
         try:
             stdout, stderr = server.communicate()
-            stdout = stdout.decode()
-            stderr = stderr.decode()
         except KeyboardInterrupt:
-            pass
+            server.send_signal(signal.SIGINT)
```

Comment:
not sure this `send_signal` is required as when we call `Job.kill()` we send the signal to the whole process GROUP, not just the process itself, so both processes should get the signal from the start.

Reply:
I'm currently unsure whether this `KeyboardInterrupt` exception can ever happen, because I can't seem to find a way to trigger it. Any ideas?

Reply:
this imo should trigger if you use a recipe like this:

```python
class MyRecipe(Recipe):
    def test(self):
        iperf_job = self.matched.m1.run(IperfServer(...), bg=True)
        sleep(5) # let it start
        iperf_job.kill(signal.SIGINT)
```

Reply:
Doesn't seem to do anything:
```py
class MyRecipe(BaseRecipe):
    host1 = HostReq()
    host1.eth0 = DeviceReq(label="xd")

    def test(self) -> None:
        m1 = self.matched.host1
        iperf_job = m1.run(IperfServer(), bg=True)
        time.sleep(5)
        iperf_job.kill(signal.SIGINT)
        time.sleep(5)
```
```
2024-01-17 13:44:57           (host1)        -   DEBUG: Running job 0 with pid "1758" ({'job_id': 0, 'type': 'module', 'json': False, 'module': IperfServer()})
2024-01-17 13:44:57           (host1)        -   DEBUG: compiled command:  iperf3 -s  -J    
2024-01-17 13:44:57           (host1)        -   DEBUG: running as server ...
2024-01-17 13:45:02       (localhost)        -   DEBUG: Sending signal 2 to job 0 on host host1
2024-01-17 13:45:02           (host1)        -   DEBUG: Sending signal 2 to pid 1758
2024-01-17 13:45:07           (host1)        -    INFO: Killing all forked processes.
2024-01-17 13:45:07           (host1)        -   DEBUG: Killing all jobs
2024-01-17 13:45:07           (host1)        -   DEBUG: Sending signal 9 to pid 1758
2024-01-17 13:45:07       (localhost)        -    INFO: Result: FAIL, What: Job finished: type(module), host(host1), netns(None), IperfServer()
2024-01-17 13:45:07           (host1)        -    INFO: Restoring system configuration
```

Reply:
hmmm, confirmed on my side as well... i think this would need some hands on debugging with the agent process via a vm as i can't easily inspect the container version of the agent process.

Reply:
Might have found the issue. When running a job, we reset `SIGINT` to `SIG_DFL` at https://github.com/LNST-project/lnst/blob/master/lnst/Agent/Job.py#L99. I tested this manually and suddenly the process stopped raising a `KeyboardInterrupt`. Apprently there is a `signal.default_int_handler`, which should be set to `SIGINT` instead...

Reply:
ah, i was looking at that yesterday actually but `SIG_DFL` should reset to the default handler and from what i remember this used to work... so maybe this has changed at some point...

Reply:
https://stackoverflow.com/a/22916853